src/client.ts

```diff
@@ -110,38 +97,23 @@ export class SubscriptionClient {
     this.client.close();
   }
 
-  public subscribe(options: SubscriptionOptions, handler: (error: Error[], result?: any) => void) {
-    const { query, variables, operationName, context } = options;
-
-    if (!query) {
-      throw new Error('Must provide `query` to subscribe.');
-    }
-
-    if (!handler) {
-      throw new Error('Must provide `handler` to subscribe.');
-    }
+  public query(options: RequestOptions): Promise<ExecutionResult> {
+    return new Promise((resolve, reject) => {
+      const handler = (error: Error[], result?: any) => {
+        if (result) {
+          resolve(result);
+        } else {
+          reject(error);
+        }
+      };
 
-    if (
-      !isString(query) ||
-      ( operationName && !isString(operationName)) ||
-      ( variables && !isObject(variables))
-    ) {
-      throw new Error('Incorrect option types to subscribe. `subscription` must be a string,' +
-      '`operationName` must be a string, and `variables` must be an object.');
-    }
+      // NOTE: as soon as we move into observables, we don't need to wait GQL_COMPLETE for queries and mutations
+      this.executeRequest(options, handler);
+    });
+  }
 
-    const subId = this.generateSubscriptionId();
-    let message = Object.assign(options, {type: SUBSCRIPTION_START, id: subId});
-    this.sendMessage(message);
-    this.subscriptions[subId] = {options, handler};
-    this.waitingSubscriptions[subId] = true;
-    setTimeout( () => {
-      if (this.waitingSubscriptions[subId]) {
-        handler([new Error('Subscription timed out - no response from server')]);
-        this.unsubscribe(subId);
-      }
-    }, this.subscriptionTimeout);
-    return subId;
+  public subscribe(options: RequestOptions, handler: (error: Error[], result?: any) => void) {
```

Comment:
here we need to add a wrapper handler for converting payload to:
```
const requestPayloadData = parsedMessage.payload.data || null;
const requestPayloadErrors = parsedMessage.payload.errors ?
  this.formatErrors(parsedMessage.payload.errors) : null;
```
to keep backward comptiable with subscription manager